File: 231/PowerOfTwo.kt
// 231. Power of Two
// https://leetcode.com/problems/power-of-two/
class Solution {
    fun isPowerOfTwo(n: Int): Boolean {
        return n > 0 && n and (n - 1) == 0
    }

    // fun isPowerOfTwo(n: Int): Boolean {
    //     var num = n
    //     while (num > 1 && (num and 1) != 1) { num = num.shr(1) }
    //     return num == 1
    // }
}

fun main(args: Array<String>) {
    val solution = Solution()
    println("=========================================")
    println(solution.isPowerOfTwo(1))
    println(true)
    println("=========================================")
    println(solution.isPowerOfTwo(16))
    println(true)
    println("=========================================")
    println(solution.isPowerOfTwo(218))
    println(false)
    println("=========================================")
    println(solution.isPowerOfTwo(0))
    println(false)
    println("=========================================")
}

File: 231/power_of_two.py
# 231. Power of Two
# https://leetcode.com/problems/power-of-two/
def main():
  solution = Solution()
  print("=========================================")
  print(solution.isPowerOfTwo(1))
  print(True)
  print("=========================================")
  print(solution.isPowerOfTwo(16))
  print(True)
  print("=========================================")
  print(solution.isPowerOfTwo(218))
  print(False)
  print("=========================================")
  print(solution.isPowerOfTwo(0))
  print(False)
  print("=========================================")

class Solution:
  def isPowerOfTwo(self, n: int) -> bool:
    return n > 0 and n & n - 1 == 0

  # def isPowerOfTwo(self, n: int) -> bool:
  #   while (n > 1 and n&1 != 1): n >>= 1
  #   return n == 1

if __name__ == '__main__':
    main()
